api/app/Models/Article.php

```diff
@@ -85,6 +87,25 @@ protected static function boot()
         });
     }
 
+    /**
+     * @param string $id article_id or permalink
+     * @return Article
+     */
+    public function findByPermalink($id)
+    {
+        //if the id is a uuid, try that or fail.
+        if (Uuid::isValid($id)) {
+            return parent::findOrFail($id);
+        }
+
+        //otherwise attempt treat the id as a permalink and first try the model, then try the history
+        try {
+            return $this->where('permalink', '=', $id)->firstOrFail();
+        } catch (ModelNotFoundException $e) { //id or permalink not found, try permalink history
+            return ArticlePermalink::findOrFail($id)->article;
+        }
+    }
+
```

Comment:
It might make more sense to have this actually override a base function `findByIdentifier($identifier)` that all controller methods use, allowing extending controllers to define their own meaning of identifier (like this one where there can be many). By default it would just be `return parent::findOrFail($id);`